rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -161,9 +186,54 @@ void whenAllActivationsArePresentAndFederationChanges_shouldCreateCommitAndActiv
         activateNewFederation(activations);
 
         // Assert
+   
         assertUTXOsReferenceMovedFromNewToOldFederation(originalUTXOs, activations);
         assertNewAndOldFederationsHaveExpectedAddress(newFederation.getAddress(), originalFederation.getAddress());
-        assertFundsWereNotMigrated();
+        assertMigrationHasNotStarted();
+    }
+
+    @Test
+    void whenAllActivationsArePresentAndFederationChanges_shouldMigrateAllFunds() throws Exception {
+        // Arrange
+    
+        var activations = ActivationConfigsForTest.all().forBlock(0);
+        setUpFederationChange(activations);
+        // Create a default original federation using the list of UTXOs
+        var originalFederation = createOriginalFederation(
+            FederationType.P2SH_ERP, ORIGINAL_FEDERATION_KEYS, activations);
+       
+        // Act & Assert
+    
+        // Create pending federation using the new federation keys
+        var newFederation = createPendingFederation(NEW_FEDERATION_KEYS, activations); 
+        commitPendingFederation();
+        // Since Lovell is activated we will commit the proposed federation
+        commitProposedFederation(activations);
+        // Move blockchain until the activation phase
+        activateNewFederation(activations);
+
+        assertNewAndOldFederationsHaveExpectedAddress(
+            newFederation.getAddress(), originalFederation.getAddress());
+        assertMigrationHasNotStarted();
+
+        // Move blockchain until the migration phase
+        activateMigration(activations);
+        // Migrate
+        migrateUTXOs();
+
+        assertNewAndOldFederationsHaveExpectedAddress(
+            newFederation.getAddress(), originalFederation.getAddress());
+        assertMigrationHasStarted();
+        verifySigHashes(activations);
+        verifyPegoutTransactionCreatedEventWasEmitted(activations);
```

Comment:
Let's add an initial check checking the state before the migration so you can verify that before the migration, there was no pegout request, waiting for confirmation or waiting for signature.

Wdyt?

Reply:
`assertMigrationHasNotStarted()` verifies that there is no pegouts in the queue. I believe it already does this.